### PR TITLE
Refactor search bar

### PR DESCRIPTION
**Pull request by cinqmilleans**
**Details of Changes**
Move the SearchRegion code to a specific control.
This simplify the code of the navigation bar.
This pr does not provide any functionality and prepares issue #4813 which improves the search bar.

### CODE REVIEW

File: Files/UserControls/NavigationToolbar.xaml [hunk 1]
```diff
@@ -834,89 +812,37 @@
...
-                Style="{StaticResource ToolBarButtonStyle}"
+                AutomationProperties.Name="Search"
                 ToolTipService.ToolTip="Search (Ctrl+F)">
                 <FontIcon FontSize="14" Glyph="{x:Bind SearchButtonGlyph, Mode=OneWay}" />
                 <Button.KeyboardAccelerators>
-                    <KeyboardAccelerator
-                        Key="F"
-                        IsEnabled="False"
-                        Modifiers="Control" />
+                    <KeyboardAccelerator Key="F" Modifiers="Control" IsEnabled="False" />
```
Comment (yaichenbaum) [inline Files/UserControls/NavigationToolbar.xaml:845]: Can you revert the formatting changes made to this button?

**Review by yaichenbaum — CHANGES_REQUESTED** [review]
The code looks really code! Can you use [XAML Styler](https://marketplace.visualstudio.com/items?itemName=TeamXavalon.XAMLStyler
) to fix the formatting on the XAML files you edited?

File: Files/UserControls/NavigationToolbar.xaml (hunk 1 above)
Comment (cinqmilleans) [reply Files/UserControls/NavigationToolbar.xaml]: @yair100 Ok, I'll do this.

Comment (yaichenbaum) [reply Files/UserControls/NavigationToolbar.xaml]: I fixed it from my end 🙂

**Review by yaichenbaum — APPROVED** [review]
LGTM

**Pull request merged**